app/transactions/admin/missions/update_transaction.rb

```diff
@@ -6,72 +6,82 @@ module Missions
     class UpdateTransaction
       include Dry::Transaction
 
-      around :rollback_if_failure
-
-      tee :remove_datetime_attributes_if_recurrent_option
+      tee :convert_datepicker_params_in_datetime
+      step :check_if_enrollment_are_inside_of_new_mission_s_period?
+      step :check_if_enrollments_match_a_mission_s_time_slots_for_regulated_mission
       step :update_mission
-      tee :get_updatable_missions
-      step :update_recurrently_other_missions
 
       private
 
-      def rollback_if_failure(input, &block)
-        result = nil
+      def convert_datepicker_params_in_datetime(input)
+        return Success(input) unless input[:params]['start_date(1i)']
 
-        Mission.transaction do
-          result = block.call(Success(input))
-          raise ActiveRecord::Rollback if result.failure?
+        start_date = convert_params_in_datetime(input[:params], 'start_date')
+        due_date = convert_params_in_datetime(input[:params], 'due_date')
+        input[:params].merge!(start_date: start_date, due_date: due_date)
+        Success(input)
+      end
 
-          result
+      def check_if_enrollment_are_inside_of_new_mission_s_period?(input)
+        enrollments = input[:mission].enrollments
+        failure_message = I18n.t('activerecord.errors.models.mission.inconsistent_datetimes_for_related_enrollments')
+        enrollments.each do |enrollment|
+          return Failure(failure_message) unless enrollment_inside_mission_s_period?(enrollment, input)
         end
-        result
+        Success(input)
       end
 
-      def remove_datetime_attributes_if_recurrent_option(input)
-        return Success(input) unless input[:params][:recurrent_change]
-
-        input[:params].delete(:start_date)
-        input[:params].delete(:due_date)
-
+      def check_if_enrollments_match_a_mission_s_time_slots_for_regulated_mission(input)
+        enrollments = input[:mission].enrollments
+        i18n_key = 'activerecord.errors.models.mission.mismatch_between_time_slots_and_related_enrollments'
+        failure_message = I18n.t(i18n_key)
+        enrollments.each do |enrollment|
+          return Failure(failure_message) unless match_a_time_slot?(input, enrollment)
+        end
         Success(input)
       end
```

Comment:
THese methods are actually validating the `Mission` object ; so move them as validation steps in the `Mission` model.
By  doing so, we might be able to get rid of this transaction object.